staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -274,6 +274,9 @@ func (s internalSelector) Empty() bool {
 // Requirement. If called on an invalid Requirement, an error is
 // returned. See NewRequirement for creating a valid Requirement.
 func (r *Requirement) String() string {
+	if r == nil {
```

Comment:
argh, github ate my comment. Given the disclaimer on this method, I don't think people should have nil requirements, and since I'm only 98% sure what you have here won't break things, I'd prefer if we reverted this one.

Reply:
You are right, people should have nil requirements. I have reverted  this change